x-pack/plugins/security/public/authentication/login/login_page.tsx

```diff
@@ -144,68 +175,120 @@ export class LoginPage extends Component<Props, State> {
+            <FormattedMessage
+              id="xpack.security.loginPage.esUnavailableMessage"
+              defaultMessage="See the Kibana logs for details and try reloading the page."
+            />
+          }
+        />
+      );
+    }
+
+    if (layout === 'error-xpack-unavailable') {
+      return (
+        <DisabledLoginForm
+          title={
+            <FormattedMessage
+              id="xpack.security.loginPage.xpackUnavailableTitle"
+              defaultMessage="Cannot connect to the Elasticsearch cluster currently configured for Kibana."
+            />
+          }
+          message={
+            <FormattedMessage
+              id="xpack.security.loginPage.xpackUnavailableMessage"
+              defaultMessage="To use the full set of free features in this distribution of Kibana, please update Elasticsearch to the default distribution."
+            />
+          }
+        />
+      );
+    }
+
+    if (layout !== 'form') {
+      return (
+        <DisabledLoginForm
+          title={
+            <FormattedMessage
+              id="xpack.security.loginPage.unknownLayoutTitle"
+              defaultMessage="Unsupported login form layout."
+            />
+          }
+          message={
+            <FormattedMessage
+              id="xpack.security.loginPage.unknownLayoutMessage"
+              defaultMessage="Refer to the Kibana logs for more details and refresh to try again."
+            />
+          }
+        />
+      );
+    }
+
+    const loginSelector =
+      showLoginSelector &&
+      selector.providers.map((provider, index) => (
+        <EuiButton
+          key={index}
+          className="loginWelcome__selectorButton"
+          iconType="user"
+          fullWidth={true}
+          onClick={() => this.login(provider.type, provider.name)}
+        >
+          {provider.options.description ?? `${provider.type}/${provider.name}`}
+        </EuiButton>
+      ));
+
+    const loginSelectorAndLoginFormSeparator = showLoginSelector && showLoginForm && (
+      <>
+        <EuiText textAlign="center" color="subdued">
+          ―――&nbsp;&nbsp;
+          <FormattedMessage id="xpack.security.loginPage.loginSelectorOR" defaultMessage="OR" />
+          &nbsp;&nbsp;―――
+        </EuiText>
+        <EuiSpacer size="m" />
+      </>
+    );
+
+    const loginForm = showLoginForm && (
+      <BasicLoginForm
+        http={this.props.http}
+        infoMessage={infoMessageMap.get(parse(window.location.href, true).query.msg?.toString())}
+        loginAssistanceMessage={this.props.loginAssistanceMessage}
+      />
+    );
+
+    return (
+      <>
+        {loginSelector}
+        {loginSelectorAndLoginFormSeparator}
+        {loginForm}
+      </>
+    );
+  };
+
+  private login = async (providerType: string, providerName: string) => {
+    try {
+      const { location } = await this.props.http.post<{ location: string }>(
+        `${this.props.http.basePath.serverBasePath}/internal/security/login_with`,
```

Comment:
I don't think so, good catch! It's a left over from previous implementation when I used links instead of buttons. Added `test with base path` to my TODO list :)